test/clj/rems/api/test_catalogue.clj

```diff
@@ -19,15 +19,47 @@
       (is (some #(str/starts-with? (:resid %) "urn:") data)))))
 
 (deftest catalogue-api-security-test
-  (testing "listing without authentication"
-    (let [response (-> (request :get (str "/api/catalogue"))
-                       handler)
-          body (read-body response)]
-      (is (response-is-unauthorized? response))
-      (is (= "unauthorized" body))))
-  (testing "listing with wrong API-Key"
-    (is (= "unauthorized"
-           (-> (request :get (str "/api/catalogue"))
-               (assoc-in [:headers "x-rems-api-key"] "invalid-api-key")
-               handler
-               (read-body))))))
+  (with-redefs [rems.config/env (assoc rems.config/env :catalogue-is-public true)]
+    (testing "catalogue-is-public true with no authentification"
+      (let [response (-> (request :get "/api/catalogue")
+                         handler)]
+        (is (response-is-ok? response))))
+    (let [api-key "42"]
+      (testing "catalogue-is-public true with authentification by api key"
```

Comment:
Maybe there is no need to check the case with api-key since this is about the public catalogue, should be enough to test the without api-key case (and at most one api-key one)?

Reply:
I don't mind the extra case

Reply:
Should I delete it then?

Reply:
your call

Reply:
I will leave it then.